lib/nostrum/cache/guild/guild_server.ex

```diff
@@ -442,4 +447,14 @@ defmodule Nostrum.Cache.Guild.GuildServer do
     {:reply, {guild_id, old_emojis, emojis}, %{state | emojis: emojis}}
   end
 
+  def handle_call({:update, :voice_state, guild_id, voice_state}, _from, state) do
+    old_voice_states = state.voice_states
+    new_voice_states =
+      [
+        voice_state |
+        state.voice_states
+        |> Enum.reject(fn(map) -> map.user_id == voice_state.user_id end)
+      ]
+    {:reply, {guild_id, old_voice_states, new_voice_states}, %{state | voice_states: new_voice_states}}
```

Comment:
As a user would it be more useful to receive a new and old list, or just the singular new and old voice state?

If we decide the latter we'll need some new logic/design choices on how to represent a voice_state being created/deleted. If the former then this looks good. I'm open to either!